docker/sui-indexer/Dockerfile

```diff
@@ -1,25 +1,50 @@
-# Build application
-#
-# Copy in all crates, Cargo.toml and Cargo.lock unmodified,
-# and build the application.
-FROM rust:1.70.0 AS builder
-ARG PROFILE=release
+FROM rust:1.70.0 AS chef
+WORKDIR sui
 ARG GIT_REVISION
 ENV GIT_REVISION=$GIT_REVISION
-WORKDIR "$WORKDIR/sui"
 RUN apt-get update && apt-get install -y cmake clang
 
+# Plan out the 3rd-party dependencies that need to be built.
+#
+# This is done by:
+#   1. Copy in Cargo.toml, Cargo.lock, and the workspace-hack crate
+#   2. Removing all workspace crates, other than the workpsace-hack
+#      crate, from the workspace Cargo.toml file.
+#   3. Update the lockfile in order to reflect the changes to the
+#      root Cargo.toml file.
+# FROM chef AS planner
+# COPY Cargo.toml Cargo.lock ./
+# COPY crates/workspace-hack crates/workspace-hack
+# RUN sed -i '/crates\/workspace-hack/b; /crates/d; /narwhal/d' Cargo.toml \
+#    && cargo metadata -q >/dev/null
+
+# Build and cache all dependencies.
+#
+# In a fresh layer, copy in the "plan" generated by the planner
+# and run `cargo build` in order to create a caching Docker layer
+# with all dependencies built.
+FROM chef AS builder 
+# COPY --from=planner /sui/Cargo.toml Cargo.toml
+# COPY --from=planner /sui/Cargo.lock Cargo.lock
+# COPY --from=planner /sui/crates/workspace-hack crates/workspace-hack
+# RUN cargo build --release
+
+# Build application
+#
+# Copy in the rest of the crates (and an unmodified Cargo.toml and Cargo.lock)
+# and build the application. At this point no dependencies should need to be
+# built as they were built and cached by the previous layer.
 COPY Cargo.toml Cargo.lock ./
 COPY crates crates
 COPY sui-execution sui-execution
 COPY narwhal narwhal
 COPY external-crates external-crates
-
-RUN cargo build --profile ${PROFILE} --bin sui-indexer
+RUN cargo build --release \
+    --bin sui-indexer
```

Comment:
This should be just one line, it's short.

Reply:
this is a revert of the PR